node/testing/src/scenarios/multi_node/pubsub_advanced.rs

```diff
@@ -20,9 +29,61 @@ use crate::{
 pub struct MultiNodePubsubPropagateBlock;
 
 impl MultiNodePubsubPropagateBlock {
-    const WORKERS: usize = 20;
+    const WORKERS: usize = 10;
 
     pub async fn run(self, mut runner: ClusterRunner<'_>) {
+        let graph = Arc::new(Mutex::new("digraph {\n".to_owned()));
+        let factory = || {
+            let graph = graph.clone();
+            move |_id,
+                  state: &node::State,
+                  _service: &NodeTestingService,
+                  action: &ActionWithMeta| {
+                let this = state.p2p.my_id();
+
+                let cut = |peer_id: &PeerId| {
+                    let st = peer_id.to_string();
+                    let len = st.len();
+                    st[(len - 6)..len].to_owned()
+                };
+                let this = cut(&this);
+
+                match action.action() {
+                    Action::P2p(P2pAction::Network(P2pNetworkAction::Pubsub(
+                        P2pNetworkPubsubAction::OutgoingMessage { peer_id },
+                    ))) => {
+                        // FIXME: this doesn't work because this is called after the action
```

Comment:
@binier @vlad9486 now that the message is not part of the state I cannot fetch it here to collect the data for the test, do you have any suggestion?

Obvious solution I can come up is to add an intermediary action before `P2pNetworkPubsubAction::OutgoingMessage` which is basically a non-op so that I can intercept that instead, but feels dirty.

Reply:
alternatively, an post-Action `OutgoingMessageClear` that clears the message there, that feels like a better option.

Reply:
I went with the second option.